### PR TITLE
Use Node Id to match topology for LR

### PR DESCRIPTION
**Pull request by zhangn49**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java [hunk 1]
```diff
@@ -864,4 +875,20 @@ private void updateReplicationStatus() {
...
+        if (nodeIdFilePath != null) {
+            File nodeIdFile = new File(nodeIdFilePath);
+            try (BufferedReader bufferedReader = new BufferedReader(new FileReader(nodeIdFile))) {
+                String line = bufferedReader.readLine();
+                localNodeId = Optional.of(line.split("=")[1].trim().toLowerCase());
+                log.info("setupLocalNodeId succeeded, node id is {}", localNodeId);
+            } catch (Exception e) {
+                log.warn("setupLocalNodeId failed", e);
+            }
+        }
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java:892]: can we add a logging warning if no Node Id file path is found?

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/LogReplicationPluginConfig.java [hunk 2]
```diff
@@ -83,6 +86,8 @@ public LogReplicationPluginConfig(String filepath) {
 
             this.topologyManagerAdapterJARPath = prop.getProperty("topology_manager_adapter_JAR_path");
             this.topologyManagerAdapterName = prop.getProperty("topology_manager_adapter_class_name");
+
+            this.nodeIdFilePath = prop.getProperty("local_node_id_path");
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/LogReplicationPluginConfig.java:90]: Can you add a comment on the method that specifies the format in which the node must be written in this file? for future reference..

**Review by annym — APPROVED** [review]

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java (hunk 1 above)
Comment (zhangn49) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java:894]: Added

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/LogReplicationPluginConfig.java (hunk 2 above)
Comment (zhangn49) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/LogReplicationPluginConfig.java:92]: Added file format : 

> node_id=99F30C42-F96D-A0A7-5531-468E52926A01

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/LogReplicationPluginConfig.java [hunk 3]
```diff
@@ -67,6 +67,11 @@
     @Getter
     private String snapshotSyncPluginCanonicalName;
 
+    // nodeId file format:
```
Comment (pankti-m) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/LogReplicationPluginConfig.java:70]: can we only say 'nodeId format'?  what does 'file' mean here?

**Review by pankti-m — APPROVED** [review]

**Pull request merged**